### PR TITLE
SAK-31819 Move ScheduledInvocationManager to directly use Quartz

### PR DESCRIPTION
**Pull request by buckett**
This registers jobs for the ScheduledInvocationManager directly in Quartz rather than having a single quartz job that runs all the ScheduledInvocationManager jobs itself. This allows better timing, concurrent execution and better reporting on ScheduledInvocationManager jobs.

Also as part of this change it add support for hibernate annotations being used rather than having to write `*.hbm.xml` files.

### CODE REVIEW

**Comment by buckett** [comment]
I'll squash all this down and make one good commit before merging but sometimes it's easier to see what's changed and why when they are separate commits.

So don't merge, but please review.

**Comment by buckett** [comment]
This moves the details of the ScheduledInvocationManager jobs between tables, and provides a Quartz job todo the conversion (SchedulerConversionJob). When starting up the SchedulerConversionJob should be run which will load all the existing ScheduledInvocationManager jobs, put them as individual quartz jobs and delete them. This only needs to be run once.

**Comment by ottenhoff** [comment]
This is pretty great @buckett .... Note that Github will do the squashing for you with Squash and merge....

**Comment by ern** [comment]
Very nice @buckett 

This is definitely moving things in the right direction.

However one more important addition could be made!
Instead of annotating the DAO's with @Transactional it would be a great improvement to annotate the service instead, this has been one of sakai's largest mistakes in regards to hibernate and transactions.

File: jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/ScheduledInvocationManagerImpl.java [hunk 1]
```diff
@@ -1,151 +1,137 @@
...
 import org.sakaiproject.time.api.Time;
 
+/**
+ * componentId -> job key (name)
+ * opaqueContent/contextId -> trigger key (name)
+ *
+ * jobs have groups and triggers have groups.
+ * matching by quartz can be done on both of them and supports equals/startswith.
+ * possiblity to have another table that does the opaqueID to UUID mapping?
+ */
```
Comment (buckett) [inline jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/ScheduledInvocationManagerImpl.java]: The implementation should be `@transactional`

**Comment by buckett** [comment]
Thanks @ern, yeah I think they should both be `@transactional`

**Comment by ern** [comment]
It depends on whether or not the DAO's can be used elsewhere. If the DAO's aren't public (which most shouldn't be) then annotating the service is correct as they would be the only users of it. But if the DAO's are public and can be used elsewhere then it does give reason to annotating the DAO, but that is probably a design flaw as the over goal should be exposing and user services.

**Comment by buckett** [comment]
The DAO isn't public and the service now has annotations.

**Comment by ern** [comment]
LGTM

**Pull request merged**